### PR TITLE
Adding Odroid N2

### PR DESCRIPTION
**Pull request by paulcrawford**
@makermelissa, I trust this is OK.

### CODE REVIEW

File: adafruit_platformdetect/board.py [hunk 1]
```diff
@@ -58,6 +58,7 @@
 ODROID_C1                   = "ODROID_C1"
 ODROID_C1_PLUS              = "ODROID_C1_PLUS"
 ODROID_C2                   = "ODROID_C2"
+ODROID_N2                   = "Hardkernel ODROID-N2"
```
Comment (makermelissa) [inline adafruit_platformdetect/board.py]: Just a little nitpick here, but maybe change the string to "ODROID_N2" so it's consistent with the other names.

Comment (ladyada) [inline adafruit_platformdetect/board.py]: match the style above

File: adafruit_platformdetect/chip.py [hunk 2]
```diff
@@ -102,6 +103,8 @@ def _linux_id(self): # pylint: disable=too-many-branches
                 linux_id = IMX8MX
             if compatible and 'odroid-c2' in compatible:
                 linux_id = S905
+            if compatible and 'amlogic, g12broot' in compatible:
```
Comment (ladyada) [inline adafruit_platformdetect/chip.py]: theres no 'odroid-n2' text?

File: adafruit_platformdetect/chip.py [hunk 3]
```diff
@@ -113,6 +116,8 @@ def _linux_id(self): # pylint: disable=too-many-branches
             linux_id = S805
         elif "ODROID-C2" in hardware:
             linux_id = S905
+        elif "Hardkernel ODROID-N2" in hardware:
```
Comment (ladyada) [inline adafruit_platformdetect/chip.py]: remove Hardkernel

**Comment by paulcrawford** [comment]
The reason I did “-N2” rather than “_N2” is that the response from the N2 to "cat /proc/device-tree/model” is "Hardkernel ODROID-N2” and I assumed that you need an exact match to detect the board. If not it can certainly be changed.

Thanks

**Comment by makermelissa** [comment]
The first one is arbitrary text and should match the variable name. The second one is an exact match, but we should remove the word "Hardkernel".

**Comment by paulcrawford** [comment]
> On Aug 5, 2019, at 17:46, ladyada <notifications@github.com> wrote:
> 
> @ladyada requested changes on this pull request.
> 
> In adafruit_platformdetect/board.py <https://github.com/adafruit/Adafruit_Python_PlatformDetect/pull/35#discussion_r310805067>:
> 
> > @@ -58,6 +58,7 @@
>  ODROID_C1                   = "ODROID_C1"
>  ODROID_C1_PLUS              = "ODROID_C1_PLUS"
>  ODROID_C2                   = "ODROID_C2"
> +ODROID_N2                   = "Hardkernel ODROID-N2"
> match the style above
> 
> 
The response is as shown:

root@no:/home/co# cat /proc/device-tree/model
Hardkernel ODROID-N2
“Hardkernel “ can be removed if finding "ODROID-N2” in the string will do.

> In adafruit_platformdetect/chip.py <https://github.com/adafruit/Adafruit_Python_PlatformDetect/pull/35#discussion_r310805256>:
> 
> > @@ -102,6 +103,8 @@ def _linux_id(self): # pylint: disable=too-many-branches
>                  linux_id = IMX8MX
>              if compatible and 'odroid-c2' in compatible:
>                  linux_id = S905
> +            if compatible and 'amlogic, g12broot' in compatible:
> theres no 'odroid-n2' text?
> 

The response is as shown:

root@no:/home/co# cat /proc/device-tree/compatible
amlogic, g12b

Sorry about having root in the response. There is no new line character in the response so “root” was just the command prompt coming back.
> In adafruit_platformdetect/chip.py <https://github.com/adafruit/Adafruit_Python_PlatformDetect/pull/35#discussion_r310805374>:
> 
> > @@ -113,6 +116,8 @@ def _linux_id(self): # pylint: disable=too-many-branches
>              linux_id = S805
>          elif "ODROID-C2" in hardware:
>              linux_id = S905
> +        elif "Hardkernel ODROID-N2" in hardware:
> remove Hardkernel
> 
> 
Again what shows up in “hardware” will be “Hardkernel ODROID-N2” so it can be removed if “ODROID-N2” alone will do.

**Comment by paulcrawford** [comment]
Not sure which is first and second, but I leave it to you to sort out what will work. I did show what the N2 responses are to the two cat commands. I am just not sure how the code treats the result. I never dealt with any Python code until last week.

Do you have an N2 in-house to test?

**Comment by makermelissa** [comment]
Hi Paul. Since you had invited me to collaborate, I went ahead and made the requested changes to your repo. Unfortunately I don't have an N2 to test on. They came out with it shortly (like a couple days) after I ordered a bunch of boards from them.

**Comment by makermelissa** [comment]
Could you please test with the changes I made @paulcrawford. If it still detects correctly, then I'll approve.

**Comment by paulcrawford** [comment]
I did make one small change: "amlogic, g12broot" became "amlogic, g12b" ("root" was part of the command prompt following the cat command.

Just saw your hyphen to underscore change so I will install again...

Everything seems to work:

co@no:~/src$ pip3 install -e git+https://github.com/paulcrawford/Adafruit_Python_PlatformDetect.git#egg=Adafruit_PlatformDetect
Obtaining Adafruit_PlatformDetect from git+https://github.com/paulcrawford/Adafruit_Python_PlatformDetect.git#egg=Adafruit_PlatformDetect
  Updating ./src/adafruit-platformdetect clone
Installing collected packages: Adafruit-PlatformDetect
  Found existing installation: Adafruit-PlatformDetect 1.2.2.dev16+g4309080
    Uninstalling Adafruit-PlatformDetect-1.2.2.dev16+g4309080:
      Successfully uninstalled Adafruit-PlatformDetect-1.2.2.dev16+g4309080
  Running setup.py develop for Adafruit-PlatformDetect
Successfully installed Adafruit-PlatformDetect
co@no:~/src$ sudo python3 blinkatest.py
Hello blinka!
Digital IO ok!
I2C ok!
SPI ok!
done!

I still do not understand why the literal "ODROID-N2" has to change to "ODROID_N2" when the actual response from the N2 is "ODROID-N2". But it works anyway and Travis is also happy.

And ODROID_N2 is an attribute of the board module:

co@no:~/src$ python3
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import adafruit_platformdetect.board
>>> dir(adafruit_platformdetect.board)
['BEAGLEBONE', 'BEAGLEBONE_AIR', 'BEAGLEBONE_BLACK', 'BEAGLEBONE_BLACK_INDUSTRIAL', 'BEAGLEBONE_BLACK_WIRELESS', 'BEAGLEBONE_BLUE', 'BEAGLEBONE_ENHANCED', 'BEAGLEBONE_GREEN', 'BEAGLEBONE_GREEN_WIRELESS', 'BEAGLEBONE_POCKETBEAGLE', 'BEAGLEBONE_POCKETBONE', 'BEAGLEBONE_USOMIQ', 'BEAGLELOGIC_STANDALONE', 'Board', 'CORAL_EDGE_TPU_DEV', 'DRAGONBOARD_410C', 'FEATHER_HUZZAH', 'FEATHER_M0_EXPRESS', 'FTDI_FT232H', 'GENERIC_LINUX_PC', 'GIANT_BOARD', 'JETSON_NANO', 'JETSON_TX1', 'JETSON_TX2', 'JETSON_XAVIER', 'NODEMCU', 'ODROID_C1', 'ODROID_C1_PLUS', 'ODROID_C2', 'ODROID_N2', 'ORANGE_PI_PC', 'ORANGE_PI_R1', 'OSD3358_DEV_BOARD', 'OSD3358_SM_RED', 'PYBOARD', 'RASPBERRY_PI_2B', 'RASPBERRY_PI_3A_PLUS', 'RASPBERRY_PI_3B', 'RASPBERRY_PI_3B_PLUS', 'RASPBERRY_PI_4B', 'RASPBERRY_PI_A', 'RASPBERRY_PI_A_PLUS', 'RASPBERRY_PI_B_PLUS', 'RASPBERRY_PI_B_REV1', 'RASPBERRY_PI_B_REV2', 'RASPBERRY_PI_CM1', 'RASPBERRY_PI_CM3', 'RASPBERRY_PI_CM3_PLUS', 'RASPBERRY_PI_ZERO', 'RASPBERRY_PI_ZERO_W', 'SIFIVE_UNLEASHED', '_BEAGLEBONE_BOARD_IDS', '_BEAGLEBONE_IDS', '_CORAL_IDS', '_JETSON_IDS', '_LINARO_96BOARDS_IDS', '_ODROID_40_PIN_IDS', '_ORANGE_PI_IDS', '_PI_REV_CODES', '_RASPBERRY_PI_40_PIN_IDS', '_RASPBERRY_PI_CM_IDS', '_SIFIVE_IDS', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'ap_chip', 'os']


So all must be well!

Thanks for all of your help.

Paul

**Pull request merged**